internal/pkg/cli/pipeline_list.go

```diff
@@ -75,60 +115,162 @@ func (o *listPipelineOpts) Ask() error {
+			Name:         pipeline.Name,
+			ManifestPath: pipeline.Path,
+			Pipeline:     cp[pipeline.Name],
+		})
+	}
+
+	b, err := json.Marshal(out)
+	if err != nil {
+		return fmt.Errorf("marshal pipelines: %w", err)
 	}
-	fmt.Fprint(o.w, out)
 
+	fmt.Fprintf(o.w, "%s\n", b)
 	return nil
 }
 
-func (o *listPipelineOpts) jsonOutput(pipelines []*codepipeline.Pipeline) (string, error) {
+// humanOutputLocal prints the name of all pipelines in the current workspace.
+func (o *listPipelineOpts) humanOutputLocal() error {
+	local, err := o.workspace.ListPipelines()
+	if err != nil {
+		return err
+	}
+
+	for _, pipeline := range local {
+		fmt.Fprintln(o.w, pipeline.Name)
+	}
+
+	return nil
+}
+
+// jsonOutputDeployed prints data about all pipelines in the given app that have been deployed.
+func (o *listPipelineOpts) jsonOutputDeployed(ctx context.Context) error {
+	pipelines, err := getDeployedPipelines(ctx, o.appName, o.pipelineLister, o.newDescriber)
+	if err != nil {
+		return err
+	}
+
 	type serializedPipelines struct {
-		Pipelines []*codepipeline.Pipeline `json:"pipelines"`
+		Pipelines []*describe.Pipeline `json:"pipelines"`
 	}
 	b, err := json.Marshal(serializedPipelines{Pipelines: pipelines})
 	if err != nil {
-		return "", fmt.Errorf("marshal pipelines: %w", err)
+		return fmt.Errorf("marshal pipelines: %w", err)
+	}
+
+	fmt.Fprintf(o.w, "%s\n", b)
+	return nil
+}
+
+// humanOutputDeployed prints the name of all pipelines in the given app that have been deployed.
+func (o *listPipelineOpts) humanOutputDeployed() error {
+	pipelines, err := o.pipelineLister.ListDeployedPipelines(o.appName)
+	if err != nil {
+		return fmt.Errorf("list deployed pipelines: %w", err)
+	}
+
+	sort.Slice(pipelines, func(i, j int) bool {
+		return pipelines[i].Name < pipelines[j].Name
+	})
+
+	for _, p := range pipelines {
+		fmt.Fprintln(o.w, p.Name)
 	}
-	return fmt.Sprintf("%s\n", b), nil
+
+	return nil
 }
 
-func (o *listPipelineOpts) humanOutput(pipelines []string) string {
-	b := &strings.Builder{}
-	for _, pipeline := range pipelines {
-		fmt.Fprintln(b, pipeline)
+func getDeployedPipelines(ctx context.Context, app string, lister deployedPipelineLister, newDescriber newPipelineDescriberFunc) ([]*describe.Pipeline, error) {
+	pipelines, err := lister.ListDeployedPipelines(app)
+	if err != nil {
+		return nil, fmt.Errorf("list deployed pipelines: %w", err)
 	}
-	return b.String()
+
+	var mux sync.Mutex
+	var res []*describe.Pipeline
+
+	g, _ := errgroup.WithContext(ctx)
+
+	for i := range pipelines {
+		pipeline := pipelines[i]
+		g.Go(func() error {
+			d, err := newDescriber(pipeline)
+			if err != nil {
+				return fmt.Errorf("create pipeline describer for %q: %w", pipeline.ResourceName, err)
+			}
+
+			info, err := d.Describe()
+			if err != nil {
+				return fmt.Errorf("describe pipeline %q: %w", pipeline.ResourceName, err)
+			}
+
+			mux.Lock()
+			defer mux.Unlock()
+			res = append(res, info.(*describe.Pipeline))
```

Comment:
What do you think of adding this to prevent scary panic:
```gp
info, ok := err.(*descrube.Pipeline)
if !ok {
	continue
}
res = append(res, info)
```

Reply:
good call, i'll spit out an error here instead